pkg/controller/controller.go

```diff
@@ -267,6 +271,12 @@ func (c *Controller) deleteK8sResources(ns, name string) error {
 		return err
 	}
 
+	// delete autoscale
+	err = utils.DeleteAutoscale(c.clientset, name, ns)
```

Comment:
I am not sure why this is needed, in theory the `OwnerReferences` should get rid of these resources when the function is deleted. Am I missing something @ngtuna ?

Reply:
The garbage collector is mainly used for cleaning the "garbage" (orphan objects that their "dad" has been removed) and we should not abuse it. If we want to delete a function, we should also actively delete all child objects. Anyway, the `OR` field actually hasn't been added to the hpa object. PR #413 is doing that.